src/ILCompiler.ReadyToRun/src/CodeGen/ReadyToRunObjectWriter.cs

```diff
@@ -58,6 +72,28 @@ public ReadyToRunObjectWriter(PEReader inputPeReader, string objectFilePath, IEn
             _inputPeReader = inputPeReader;
         }
 
+        /// <summary>
+        /// Determine OS override for the active hosting OS.
+        /// TODO: This will need changing once we enable cross-OS build
```

Comment:
This should use `_nodeFactory.Target.OperatingSystem`. We already have the logic that sets it up based on RuntimeInformation with a TODO to make it overridable.

Reply:
Great suggestion, that is indeed much simpler and cleaner, thank you! Fixed in 3rd commit.